4.0.0 build failure

REF: https://github.com/Homebrew/homebrew-core/pull/19948

For a full log see https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/11773/version=sierra/consoleText
(duplicate in case that gets deleted https://gist.github.com/ilovezfs/82ff5572a74b6e14871c3f5af3ab352e)

The error is
```
clang    -pipe -std=gnu99 -Iinclude/ -Iinclude/lzma_sdk/ -IOpenCL/ -O2 src/main.c obj/affinity.NATIVE.STATIC.o obj/autotune.NATIVE.STATIC.o obj/benchmark.NATIVE.STATIC.o obj/bitmap.NATIVE.STATIC.o obj/bitops.NATIVE.STATIC.o obj/combinator.NATIVE.STATIC.o obj/common.NATIVE.STATIC.o obj/convert.NATIVE.STATIC.o obj/cpt.NATIVE.STATIC.o obj/cpu_aes.NATIVE.STATIC.o obj/cpu_crc32.NATIVE.STATIC.o obj/cpu_des.NATIVE.STATIC.o obj/cpu_md4.NATIVE.STATIC.o obj/cpu_md5.NATIVE.STATIC.o obj/cpu_sha1.NATIVE.STATIC.o obj/cpu_sha256.NATIVE.STATIC.o obj/debugfile.NATIVE.STATIC.o obj/dictstat.NATIVE.STATIC.o obj/dispatch.NATIVE.STATIC.o obj/dynloader.NATIVE.STATIC.o obj/event.NATIVE.STATIC.o obj/ext_ADL.NATIVE.STATIC.o obj/ext_nvapi.NATIVE.STATIC.o obj/ext_nvml.NATIVE.STATIC.o obj/ext_OpenCL.NATIVE.STATIC.o obj/ext_sysfs.NATIVE.STATIC.o obj/ext_xnvctrl.NATIVE.STATIC.o obj/ext_lzma.NATIVE.STATIC.o obj/lzma_sdk/Alloc.NATIVE.STATIC.o obj/lzma_sdk/Lzma2Dec.NATIVE.STATIC.o obj/lzma_sdk/LzmaDec.NATIVE.STATIC.o obj/filehandling.NATIVE.STATIC.o obj/folder.NATIVE.STATIC.o obj/hashcat.NATIVE.STATIC.o obj/hashes.NATIVE.STATIC.o obj/hlfmt.NATIVE.STATIC.o obj/hwmon.NATIVE.STATIC.o obj/induct.NATIVE.STATIC.o obj/interface.NATIVE.STATIC.o obj/locking.NATIVE.STATIC.o obj/logfile.NATIVE.STATIC.o obj/loopback.NATIVE.STATIC.o obj/memory.NATIVE.STATIC.o obj/monitor.NATIVE.STATIC.o obj/mpsp.NATIVE.STATIC.o obj/opencl.NATIVE.STATIC.o obj/outfile_check.NATIVE.STATIC.o obj/outfile.NATIVE.STATIC.o obj/pidfile.NATIVE.STATIC.o obj/potfile.NATIVE.STATIC.o obj/restore.NATIVE.STATIC.o obj/rp.NATIVE.STATIC.o obj/rp_cpu.NATIVE.STATIC.o obj/rp_kernel_on_cpu.NATIVE.STATIC.o obj/rp_kernel_on_cpu_optimized.NATIVE.STATIC.o obj/selftest.NATIVE.STATIC.o obj/shared.NATIVE.STATIC.o obj/status.NATIVE.STATIC.o obj/stdout.NATIVE.STATIC.o obj/straight.NATIVE.STATIC.o obj/terminal.NATIVE.STATIC.o obj/thread.NATIVE.STATIC.o obj/timer.NATIVE.STATIC.o obj/tuningdb.NATIVE.STATIC.o obj/usage.NATIVE.STATIC.o obj/user_options.NATIVE.STATIC.o obj/wordlist.NATIVE.STATIC.o                -s -framework OpenCL -lpthread -liconv -DCOMPTIME=1509128756 -DVERSION_TAG=\"HEAD -> master, tag: v4.0.0\" -DINSTALL_FOLDER=\"/usr/local/Cellar/hashcat/4.0.0/bin\" -DSHARED_FOLDER=\"/usr/local/Cellar/hashcat/4.0.0/share/hashcat\" -DDOCUMENT_FOLDER=\"/usr/local/Cellar/hashcat/4.0.0/share/doc/hashcat\" -o hashcat
clang: error: -E or -x required when input is from standard input
clang: error: no such file or directory: 'tag:'
clang: error: no such file or directory: 'v4.0.0"'
make: *** [hashcat] Error 1
```

It seems that
```
    inreplace "src/Makefile", '| $(SED) \'s/.*: v\([\.0-9]*\),.*/v\1/\')',
                              '| $(SED) \'s/.*: v\([\.0-9]*\),\?.*/v\1/\')'
```
is an effective workaround in the Homebrew formula (courtesy of @JCount)

--- a/src/Makefile
+++ b/src/Makefile
@@ -127,7 +127,7 @@ COMPTIME                := $(shell date +%s)
 # the value will be something like this: "tag: vX.Y.Z, refs/pull/Y/head"
 
 VERSION_EXPORT          := $Format:%D$
-VERSION_TAG             := $(shell test -d .git && git describe --tags --dirty=+ || echo "$(VERSION_EXPORT)" | $(SED) 's/.*: v\([\.0-9]*\),.*/v\1/')
+VERSION_TAG             := $(shell test -d .git && git describe --tags --dirty=+ || echo "$(VERSION_EXPORT)" | $(SED) 's/.*: v\([\.0-9]*\).*/v\1/')
 
 ##
 ## General compiler and linker options

